bapsflib/_hdf/maps/controls/tests/fauxnixyz.py

```diff
@@ -0,0 +1,311 @@
+            NN.append((Nx, Ny, Nz))
+
+        # add motionlist sub-groups
+        # - define motionlist names
+        # - create motionlist group
+        # - define motionlist group attributes
+        for i in range(self.knobs.n_motionlists):
+            # define motionlist name
+            ml_name = 'ml-{:04}'.format(i + 1)
+            self._ml.append({'name': ml_name,
+                             'size': sn_size_for_ml[i]})
+            # self._motionlist_names.append(ml_name)
+
+            # create motionlist group
+            ml_gname = ml_name
+            self.create_group(ml_gname)
+
+            # set motionlist attributes
+            self[ml_gname].attrs.update({
+                'z_port': np.float64(35.0),
+                'x0': np.float64(0.0),
+                'Nx': np.float64(NN[i][0]),
+                'dx': np.float64(0.5),
+                'y0': np.float64(0.0),
+                'Ny': np.float64(NN[i][1]),
+                'dy': np.float64(0.5),
+                'z0': np.float64(0.0),
+                'Nz': np.float64(NN[i][2]),
+                'dz': np.float64(0.5),
+                'fan_XYZ':
+                    np.bytes_('FALSE') if bool(i % 2)
+                    else np.bytes_('TRUE'),
+                'min_zdrive_steps': np.float64(16491),
+                'max_zdrive_steps': np.float64(-22833),
+                'min_ydrive_steps': np.float64(16491),
+                'max_ydrive_steps': np.float64(-22833),
+                'probe_name': np.bytes_('probe1'),
+            })
+
+        # fill configs dict
+        # there's one config and all ml's are in it
+        for config_name in self.configs:
+            self.configs[config_name]['motion lists'] = \
+                [self._ml[i]['name']
+                 for i in range(self.knobs.n_motionlists)]
+            # self._configs[config_name]['motion lists'] = \
+            #     self._motionlist_names
+
+    def _set_xyz_attrs(self):
+        """Set the 'NI_XYZ' group attributes"""
+        self.attrs.update({
+            'Device name': np.bytes_('NI_XYZ'),
+            'Type': np.bytes_('Experimental control'),
+            'Description': np.bytes_(
+                'The NI_XYZ device provides accurate probe position '
+                'control on the central horizontal plane on LAPD.\n'
+                'Uses National Instruments motion hardware.'),
+            'Module IP address': np.bytes_('192.168.7.32'),
+            'Module VI path': np.bytes_('Modules\\NI_XYZ\\NI_XYZ.vi'),
+            'Created data': np.bytes_('1/28/2015 3:03:58 PM'),
+        })
+
+    @property
+    def knobs(self):
+        """Knobs for controlling structure of device group"""
+        return self._knobs
+
+    def populate(self):
+        """
+        Updates the control group structure (Groups, Datasets, and
+        Attributes)
+        """
+        # clear group before rebuild
+        self.clear()
+
+        # re-initialize key lists
+        self._ml = []
```

Comment:
improve initialization and clearing of attributes (2 of 3)

```suggestion
        self._ml.clear()
```